lib/functions/rootfs/distro-specific.sh

```diff
@@ -44,25 +41,21 @@ function install_distribution_specific() {
 		install_artifact_deb_chroot "armbian-base-files"
 	fi
 
-	# Basic Netplan config. Let NetworkManager/networkd manage all devices on this system
-	if [[ -d "${SDCARD}"/etc/netplan ]]; then
-
-		declare RENDERER=networkd
-		if [ -d "${SDCARD}"/etc/NetworkManager ]; then
-			local RENDERER=NetworkManager
-		fi
-
-		cat <<- EOF > "${SDCARD}"/etc/netplan/armbian-default.yaml
-		network:
-		  version: 2
-		  renderer: ${RENDERER}
-		EOF
-	fi
-
 	# Set DNS server if systemd-resolved is in use
 	if [[ -n "$NAMESERVER" && -f "${SDCARD}"/etc/systemd/resolved.conf ]]; then
-		sed -i "s/#DNS=.*/DNS=$NAMESERVER/g" "${SDCARD}"/etc/systemd/resolved.conf
 		display_alert "Setup DNS server for systemd-resolved" "${NAMESERVER}" "info"
+
+		# Use resolved.conf.d/ directory as recommended by resolved itself
+		mkdir -p "${SDCARD}"/etc/systemd/resolved.conf.d/
+
+		cat <<- EOF > "${SDCARD}"/etc/systemd/resolved.conf.d/00-armbian-default-dns.conf
+			# Added by Armbian
+			#
+			# See resolved.conf(5) for details
+
+			[Resolve]
+			DNS=${NAMESERVER}
+		EOF
```

Comment:
Sorry for the necro @ColorfulRhino but I've been badly bit by this. This seems to force resolved to use $NAMESERVER (which is _always_ in practice set to `1.0.0.1`) despite anything obtained through DHCP. I don't think we should have this in at all -- $NAMESERVER is required during _image building_ but should be left to DHCP or manual user configuration at runtime. Why did you introduce this?

Reply:
Indeed. So the behavior is the same -- what changed is that MINIMAL images actually started using systemd-networkd and systemd-resolved (and so did my cloud-images, which is how I found it). Thanks for the investigation.